runtime/src/accounts_db.rs

```diff
@@ -184,11 +187,12 @@ impl Versioned for (u64, AccountInfo) {
 
 pub enum LoadedAccountAccessor<'a> {
     Stored(Option<(Arc<AccountStorageEntry>, usize)>),
-    Cached((&'a AccountsCache, Slot, &'a Pubkey)),
+    // None value in Cached variant means the cache was flushed
+    Cached(Option<(Pubkey, Cow<'a, CachedAccount>)>),
 }
 
 impl<'a> LoadedAccountAccessor<'a> {
-    fn get_loaded_account(&self) -> Option<LoadedAccount> {
+    fn get_loaded_account(&mut self) -> Option<LoadedAccount> {
         match self {
             LoadedAccountAccessor::Stored(storage_entry) => {
                 // May not be present if slot was cleaned up in between
```

Comment:
nits: how about documenting our learnings?

```rust
// This not-deterministic behavior can't happen via consensus-sensitive tx execution path (like `load()`) but can do via rpc.
```

Reply:
@ryoqun ha! I think there is actually an additional edge case to account for, fixed here: https://github.com/solana-labs/solana/pull/15454/commits/e7ae7620f23cf849b16efa621f5498f0ef7813c1

Reply:
Reminder: I should write additional test case for this...